projects/ngrx.io/content/guide/signals/signal-method.md

```diff
@@ -0,0 +1,200 @@
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly logDoubledNumber = signalMethod<number>(num => {
+    const double = num * 2;
+    console.log(double);
+  })
+
+  ngOnInit(): void {
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 uses the injection context of the `signalMethod`
+    this.logDoubledNumber(value);
+  }
+}
+```
+
+For the `NumbersComponent`, it doesn't make a difference. Again, the `effect` is automatically cleaned up when the component is destroyed.
+
+Careful, when `signalMethod` is used in a service which is provided in `root`:
+
+```ts
+
+@Injectable({providedIn: 'root'})
+export class NumbersService {
+  readonly logDoubledNumber = signalMethod<number>(num => {
+    const double = num * 2;
+    console.log(double);
+  })
+}
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly logDoubledNumber = inject(NumbersService).logDoubledNumber;
+
+  ngOnInit(): void {
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 uses the injection context of the `NumbersService`, which is root.
+    this.logDoubledNumber(value);
+  }
+}
+```
+
+Here, the `effect` outlives the component, which would produce a memory leak.
+
+As a consequence, try to call the "processor function" always in an injection context:
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly logDoubledNumber = inject(NumbersService).logDoubledNumber;
+
+  ngOnInit(): void {
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 uses the injection context of the `NumbersService`, which is root.
+    this.logDoubledNumber(value);
+  }
+}
+```
+
+## Providing an Injector
+
+If you cannot run the "processor function" in an injection context, you can also provide an injector manually:
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly logDoubledNumber = inject(NumbersService).logDoubledNumber;
+  readonly injector = inject(Injector);
+
+  ngOnInit(): void {
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 uses the injection context of the `NumbersService`, which is root.
+    this.logDoubledNumber(value, {injector: this.injector});
+  }
+}
+```
+
+Whereas the "processor function" doesn't require an active injection context, the call of `signalMethod` does:
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+// 👇 Would cause a runtime error
+  ngOnInit() {
+    const logDoubledNumber = signalMethod<number>(num => console.log(num * 2));
+  }
+}
+```
+
+In these cases, you also have to provide the injector manually:
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly injector = inject(Injector);
+
+  ngOnInit() {
+    // 👇 Works now
+    const logDoubledNumber = signalMethod<number>(num => console.log(num * 2), {injector: this.injector});
+  }
+}
+```
```

Comment:
This can be a separate section:

```suggestion
## Initialization Outside of Injection Context

The `signalMethod` must be initialized within an injection context. To initialize it outside an injection context, it's necessary to provide an injector as the second argument:

```ts
@Component({ /* ... */ })
export class NumbersComponent implements OnInit {
  readonly injector = inject(Injector);

  ngOnInit() {
    const logDoubledNumber = signalMethod<number>(
      (num) => console.log(num * 2),
      { injector: this.injector },
    );
  }
}
```
```